### PR TITLE
add tutorial for Bedrock Guardrails

### PR DESCRIPTION
**Pull request by jngz-es**
### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
+# Topic
+
+> Remote guardrails is released in OpenSearch 2.15. To read more details, see the OpenSearch document [Guardrails](https://opensearch.org/docs/latest/ml-commons-plugin/remote-models/guardrails/).
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:3]:
```suggestion
> Remote model guardrails has been released in OpenSearch 2.15. For more information, see [Guardrails](https://opensearch.org/docs/latest/ml-commons-plugin/remote-models/guardrails/).
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+This tutorial demonstrates how to apply your Bedrock Guardrails for your remote models in two ways.
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:5]:
```suggestion
This tutorial demonstrates how to apply your Amazon Bedrock guardrails to your externally hosted models in two ways.
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+Note: Replace the placeholders starting with the prefix `your_` with your own values.
+
+# Use the Bedrock Guardrails Independent API
+
+## 0. preparation
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:11]:
```suggestion
## 0. Preparation
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+Create your own Bedrock Guardrails. For more details, see the Bedrock document [Create a guardrail](https://docs.aws.amazon.com/bedrock/latest/userguide/guardrails-create.html).
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:12]:
```suggestion
Create your own Amazon Bedrock guardrails. For more information, see [Create a guardrail](https://docs.aws.amazon.com/bedrock/latest/userguide/guardrails-create.html).
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+## 1. Create a connector with your Bedrock Guardrails endpoint
+
+An example command:
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:16]:
```suggestion
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+      },
+      "request_body": "{\"source\":\"${parameters.source}\", \"content\":[ { \"text\":{\"text\": \"${parameters.input}\"} } ] }"
+    }
+  ]
+}
+```
+
+## 2. Create a remote model with your guardrail connector
+
+An example command:
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:50]:
```suggestion
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+POST _plugins/_ml/models/_register
+{
+  "name": "bedrock test guardrails API",
+  "function_name": "remote",
+  "description": "guardrails test model",
+  "connector_id": "your_connector_id"
+}
+```
+
+## 3. Model inference
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:61]:
```suggestion
## 3. Test the model
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+
+An example command:
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:63]:
```suggestion
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+          }
+        }
+      ],
+      "status_code": 200
+    }
+  ]
+}
+```
+
+# Use the guardrail embedded in Bedrock model inference API
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:137]:
```suggestion
# Use the guardrails embedded in the Amazon Bedrock Model Inference API
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+
+## 0. preparation
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:139]:
```suggestion
## 0. Preparation
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+Create your own Bedrock Guardrails. For more details, see the Bedrock document [Create a guardrail](https://docs.aws.amazon.com/bedrock/latest/userguide/guardrails-create.html).
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:140]:
```suggestion
Create your own Amazon Bedrock guardrails. For more information, see [Create a guardrail](https://docs.aws.amazon.com/bedrock/latest/userguide/guardrails-create.html).
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+
+## 1. Create a connector of Bedrock model with guardrails headers
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:142]:
```suggestion
## 1. Create a connector for an Amazon Bedrock model with guardrail headers
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+
+An example command:
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:144]:
```suggestion
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 14]
```diff
@@ -0,0 +1,246 @@
...
+        "X-Amzn-Bedrock-GuardrailIdentifier": "your_GuardrailIdentifier",
+        "X-Amzn-Bedrock-GuardrailVersion": "1"
+      },
+      "request_body": "{\"prompt\":\"${parameters.prompt}\", \"max_tokens_to_sample\":${parameters.max_tokens_to_sample}, \"temperature\":${parameters.temperature},  \"anthropic_version\":\"${parameters.anthropic_version}\" }",
+      "post_process_function": "\n      if (params['amazon-bedrock-guardrailAction']=='INTERVENED') throw new IllegalArgumentException(\"test guardrail from post process function\");\n    "
+    }
+  ]
+}
+```
+A `post_process_function` is required to define the logic for the input blocked by the guardrail or not.
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:185]:
```suggestion
A `post_process_function` is required to define the logic for blocking the input by the guardrail.
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 15]
```diff
@@ -0,0 +1,246 @@
...
+
+## 2. Create a model
+
+An example command:
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:189]:
```suggestion
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 16]
```diff
@@ -0,0 +1,246 @@
...
+POST _plugins/_ml/models/_register
+{
+  "name": "bedrock model with guardrails",
+  "function_name": "remote",
+  "description": "guardrails test model",
+  "connector_id": "your_connector_id"
+}
+```
+
+## 3. Model inference
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:200]:
```suggestion
## 3. Test the model
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 17]
```diff
@@ -0,0 +1,246 @@
...
+
+An example command:
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:202]:
```suggestion
```

**Review by kolchfa-aws — COMMENTED** [review]
Some comments

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 18]
```diff
@@ -0,0 +1,246 @@
...
+{
+  "name": "BedRock claude Connector",
+  "description": "BedRock claude Connector",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "parameters": {
+      "region": "us-east-1",
+      "service_name": "bedrock",
+      "anthropic_version": "bedrock-2023-05-31",
+      "endpoint": "bedrock.us-east-1.amazonaws.com",
```
Comment (ylwu-amzn) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:156]: This parameter not being used, delete it ?

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 19]
```diff
@@ -0,0 +1,246 @@
...
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "BedRock claude Connector",
+  "description": "BedRock claude Connector",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "parameters": {
+      "region": "us-east-1",
+      "service_name": "bedrock",
+      "anthropic_version": "bedrock-2023-05-31",
```
Comment (ylwu-amzn) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:155]: Delete this parameter as this not being used ?

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 20]
```diff
@@ -0,0 +1,246 @@
...
+      "endpoint": "bedrock.us-east-1.amazonaws.com",
+      "auth": "Sig_V4",
+      "content_type": "application/json",
```
Comment (ylwu-amzn) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:158]: remove these too

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 21]
```diff
@@ -0,0 +1,236 @@
...
+
+```
+POST _plugins/_ml/connectors/_create
+{
+  "name": "BedRock claude Connector",
+  "description": "BedRock claude Connector",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "parameters": {
+    "region": "us-east-1",
```
Comment (ylwu-amzn) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:24]:
```suggestion
    "region": "your_aws_region like us-east-1",
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 22]
```diff
@@ -0,0 +1,236 @@
...
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-v2/invoke",
+      "headers": { 
+        "content-type": "application/json",
+        "x-amz-content-sha256": "required",
+        "X-Amzn-Bedrock-Trace": "ENABLED",
+        "X-Amzn-Bedrock-GuardrailIdentifier": "your_GuardrailIdentifier",
+        "X-Amzn-Bedrock-GuardrailVersion": "1"
```
Comment (ylwu-amzn) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:169]:
```suggestion
        "X-Amzn-Bedrock-GuardrailVersion": "your_bedrock_guardrail_version"
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 23]
```diff
@@ -0,0 +1,236 @@
...
+
+```
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "BedRock claude Connector",
+  "description": "BedRock claude Connector",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "parameters": {
+      "region": "us-east-1",
```
Comment (ylwu-amzn) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:149]:
```suggestion
      "region": "your_aws_region like us-east-1",
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 24]
```diff
@@ -0,0 +1,236 @@
...
+      "url": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-v2/invoke",
+      "headers": { 
+        "content-type": "application/json",
+        "x-amz-content-sha256": "required",
+        "X-Amzn-Bedrock-Trace": "ENABLED",
+        "X-Amzn-Bedrock-GuardrailIdentifier": "your_GuardrailIdentifier",
+        "X-Amzn-Bedrock-GuardrailVersion": "1"
+      },
+      "request_body": "{\"prompt\":\"${parameters.prompt}\", \"max_tokens_to_sample\":${parameters.max_tokens_to_sample}, \"temperature\":${parameters.temperature},  \"anthropic_version\":\"${parameters.anthropic_version}\" }",
+      "post_process_function": "\n      if (params['amazon-bedrock-guardrailAction']=='INTERVENED') throw new IllegalArgumentException(\"test guardrail from post process function\");\n    "
```
Comment (ylwu-amzn) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:172]: @jngz-es I think we should enhance this part , like add default process function

**Comment by ylwu-amzn** [comment]
Suggest also add how to use the guardrail model for `input_guardrail` `output_guardrail`, like this test example 
```
POST /_plugins/_ml/models/_register?deploy=true
{
    "name": "Bedrock Claude V2 model with openai guardrails model",
    "function_name": "remote",
    "model_group_id": "ppSmpo8Bi-GZ0tf1i7cD",
    "description": "Bedrock Claude V2 model with openai guardrails model",
    "connector_id": "xnJjDZABNFJeYR3IPvTO",
    "guardrails": {
        "input_guardrail": {
            "model_id": "o3JaDZABNFJeYR3I2fRV",
            "response_accept": "^\\s*\"[Aa]ccept\"\\s*$"
        },
        "output_guardrail": {
            "model_id": "o3JaDZABNFJeYR3I2fRV",
            "response_accept": "^\\s*\"[Aa]ccept\"\\s*$"
        },
        "type": "model"
    }
}
```

File: docs/tutorials/guardrails/use_bedrock_guardrails.md (hunk 24 above)
Comment (jngz-es) [reply docs/tutorials/guardrails/use_bedrock_guardrails.md:282]: Yeah, agreed. I can add a default process function in next release version.

**Comment by jngz-es** [comment]
@kolchfa-aws , could you help have another review?

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 25]
```diff
@@ -0,0 +1,346 @@
...
+## 6. Test the model
+
+```
+POST /_plugins/_ml/models/your_model_id/_predict
+{
+  "parameters": {
+    "prompt": "\n\nHuman:${parameters.question}\n\nnAssistant:",
+    "question": "hello"
+  }
+}
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:200]:
````suggestion
}
```
````

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 26]
```diff
@@ -0,0 +1,346 @@
...
+
+Response
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:202]:
````suggestion
Response:

```
````

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 27]
```diff
@@ -0,0 +1,346 @@
...
+          "name": "response",
+          "dataAsMap": {
+            "response": " Hello!"
+          }
+        }
+      ],
+      "status_code": 200
+    }
+  ]
+}
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:217]:
````suggestion
}
```

````

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 28]
```diff
@@ -0,0 +1,346 @@
...
+
+POST /_plugins/_ml/models/your_model_id/_predict
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:219]:
````suggestion
```
POST /_plugins/_ml/models/your_model_id/_predict
````

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 29]
```diff
@@ -0,0 +1,346 @@
...
+{
+  "parameters": {
+    "prompt": "\n\nHuman:${parameters.question}\n\nnAssistant:",
+    "question": "how to suicide"
+  }
+}
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:225]:
````suggestion
}
```
````

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 30]
```diff
@@ -0,0 +1,346 @@
...
+
+Response
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:227]:
````suggestion
Response:

```
````

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 31]
```diff
@@ -0,0 +1,346 @@
...
+        "type": "illegal_argument_exception",
+        "reason": "guardrails triggered for user input"
+      }
+    ],
+    "type": "illegal_argument_exception",
+    "reason": "guardrails triggered for user input"
+  },
+  "status": 400
+}
+
```
Comment (kolchfa-aws) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:241]:
```suggestion
```

**Review by kolchfa-aws — APPROVED** [review]
A couple of formatting changes, otherwise LGTM

**Review by mingshl — APPROVED** [review]

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 32]
```diff
@@ -0,0 +1,354 @@
...
+}
+```
+
+## 3. Test the model
+
+```
+POST _plugins/_ml/models/your_model_id/_predict
+{
+  "parameters": {
+    "input": "\n\nHuman:how to suicide\n\nAssistant:"
```
Comment (mingshl) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:315]: wondering if the test sentences is safe to publish on github? probably this is question for @kolchfa-aws

File: docs/tutorials/guardrails/use_bedrock_guardrails.md [hunk 33]
```diff
@@ -0,0 +1,354 @@
...
+POST _plugins/_ml/connectors/_create
+{
+  "name": "BedRock Guardrail Connector",
+  "description": "BedRock Guardrail Connector",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "parameters": {
+    "region": "your_aws_region like us-east-1",
+    "service_name": "bedrock",
+    "source": "INPUT"
```
Comment (dhrubo-os) [inline docs/tutorials/guardrails/use_bedrock_guardrails.md:26]: out of curiosity, what is this value for?

**Review by Zhangxunmt — APPROVED** [review]

**Pull request merged**